src/widgets/mappainter.cpp

```diff
@@ -28,15 +28,32 @@ MapPainter::~MapPainter()
     delete m_beacon;
 }
 
+/**
+ * Set's distcoveredEdges list so that the
+ * painter can highlight the discovered path
+ * alternate container to `m_discoveredEdgesVector`
+ *
+ * @param discoveredEdges list of completed edges
+ */
+void MapPainter::setDiscoveredList(const std::list<Database::completedEdge>& discoveredEdges)
+{
+    /* If alternate list is empty then set vector */
+    if(m_discoveredEdgesVector.empty())
```

Comment:
Why are we checking the discovered vector here?

Reply:
making it so that only one container can be set at a time, because there are two possible ways to draw highlighted edges

Reply:
Also, could we just overload the function?

Reply:
Yes